solidity/test/KeepFactorySelectionTest.js

```diff
@@ -212,48 +270,192 @@ describe("KeepFactorySelection", async () => {
+    })
 
-      await expectRevert(
-        keepFactorySelection.setFullyBackedKeepFactory(ethStakeFactory.address),
-        "Fully backed factory already set",
+    // KEEP stake factory set.
+    // ETH-only factory not set.
+    it("updates value in KEEP stake factory", async () => {
+      await keepFactorySelection.setMinimumBondableValue(newValue, 5, 3)
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(newValue)
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(
+        defaultValue,
       )
     })
 
-    it("can not be called for 0 address", async () => {
-      await expectRevert(
-        keepFactorySelection.setFullyBackedKeepFactory(constants.ZERO_ADDRESS),
-        "Invalid address",
+    // KEEP stake factory set.
+    // ETH-only factory set.
+    it("updates values in KEEP stake and ETH-only factories", async () => {
+      await keepFactorySelection.setFactories(
+        keepStakedFactory.address,
+        fullyBackedFactory.address,
+        constants.ZERO_ADDRESS,
       )
+
+      await keepFactorySelection.setMinimumBondableValue(newValue, 5, 3)
+
+      expect(await keepStakedFactory.minimumBondableValue()).to.eq.BN(newValue)
+      expect(await fullyBackedFactory.minimumBondableValue()).to.eq.BN(newValue)
     })
   })
 
-  describe("setKeepFactorySelector", async () => {
-    it("can be called only one time", async () => {
-      await keepFactorySelection.setKeepFactorySelector(
+  describe("setFactories", async () => {
+    it("can be called multiple times", async () => {
+      const newKeepStakedFactory = await ECDSAKeepFactoryStub.new()
+      const newFullyBackedFactory = await ECDSAKeepFactoryStub.new()
+      const newSelector = await KeepFactorySelectorStub.new()
+
+      await keepFactorySelection.setFactories(
+        keepStakedFactory.address,
+        fullyBackedFactory.address,
         keepFactorySelector.address,
       )
-      // ok, this was the first time
 
-      await expectRevert(
-        keepFactorySelection.setKeepFactorySelector(
-          keepFactorySelector.address,
-        ),
-        "Factory selector already set",
+      assert.include(
+        await keepFactorySelection.factories(),
+        {
+          _keepStakedFactory: keepStakedFactory.address,
+          _fullyBackedFactory: fullyBackedFactory.address,
+          _factorySelector: keepFactorySelector.address,
+        },
+        "invalid factories after first update",
+      )
+
+      await keepFactorySelection.setFactories(
+        newKeepStakedFactory.address,
+        newFullyBackedFactory.address,
+        newSelector.address,
+      )
+
+      assert.include(
+        await keepFactorySelection.factories(),
+        {
+          _keepStakedFactory: newKeepStakedFactory.address,
+          _fullyBackedFactory: newFullyBackedFactory.address,
+          _factorySelector: newSelector.address,
+        },
+        "invalid factories after second update",
```

Comment:
Pretty weird to swap to `assert` here. `expect(...).to.include(...)` does the same thing here while sticking to our preferred mocha expectation style, right? If not, we can keep it as is, just wanted to check.

Reply:
Totally agree! I was experimenting with some other verification functions from `assert` that are not supported by `expect`, but finally ended up with `include` that is available in `expect`. I'll update this according to the suggestion.